Linked List Cycle/Main.cs

```diff
@@ -0,0 +1,35 @@
+﻿
+/**
+ * Definition for singly-linked list.
+ * public class ListNode {
+ *     public int val;
+ *     public ListNode next;
+ *     public ListNode(int x) {
+ *         val = x;
+ *         next = null;
+ *     }
+ * }
+ */
+public class Solution {
+    public bool HasCycle(ListNode head) {
+        //no nodes
+        if (head == null) return false;
+        //one node
+        if (head.next == null) return false;
+        ListNode x = head;
+        ListNode y = head.next;
+        while (y!=null)
+        {
+            if (x == y) return true;
+            //one step
+            if (x.next != null) x = x.next;
```

Comment:
Let's have the if body in a separate line, to make it more readable.
if (x.next != null) 
 x = x.next;

Reply:
We can use early return here

Reply:
hmmm yeah by inverting the if statements, good note thanks